# DoodleVPN referral sprint — 31 July 2026

## tl;dr

The scalable supply is not the exhausted audiences of a few acquaintances. It is the mostly inactive personal graph of the ordinary user base: only 6.6% of 1,144 active payers invited anyone in the last 30 days, while ordinary referred users convert to paid at about 35.1% in seven days.

The recommended 30-day treatment is one simple gift: **the friend gets seven trial days instead of three; the referrer gets 30 subscription days after the friend's first verified payment**. The existing 20% obligation stays during the experiment so the test can launch without rewriting old contracts. If it wins, new referrals should choose either days or money rather than receiving both forever.

## Context & Methods

### Key Assumptions

- Business data covers flows since 13 April 2026 and closes July at 30 July.
- The scenario model targets ordinary active payers and excludes special partners and large outliers.
- Incremental inviter rates are assumptions, not forecasts.
- 30-day revenue per ordinary referred payer is $6.98.
- Opportunity cost treats every 30-day reward as if it fully displaced a one-month renewal worth $3.13; this is deliberately conservative.
- The reward's infrastructure cash cost is modeled separately at $0.24 per user-month.

In [1]:
import csv
from pathlib import Path

ROOT = Path('analysis/doodlevpn-referral-sprint-2026-07-31')

def read_csv(name):
    with (ROOT / name).open(encoding='utf-8') as handle:
        return list(csv.DictReader(handle))

scenarios = read_csv('sprint_scenarios.csv')
facts = read_csv('referral_facts.csv')

## Data

In [2]:
facts

[{'metric': 'Active payers', 'value': '1144', 'interpretation': 'Addressable satisfied-user base'}, {'metric': 'Invited anyone in last 30d', 'value': '75', 'interpretation': 'Only 6.6% of active payers'}, {'metric': 'Never invited anyone', 'value': '821', 'interpretation': '71.8% of active payers'}, {'metric': 'Zero active paid referrals', 'value': '960', 'interpretation': 'Largest untapped group'}, {'metric': 'Stopped after 1–2 registrations', 'value': '441', 'interpretation': '72.4% of all historical referrers'}, {'metric': 'Ordinary referral 7d paid conversion', 'value': '0.351', 'interpretation': 'High quality after a friend arrives'}]

In [3]:
scenarios

[{'scenario': 'Pessimistic', 'target_inviter_rate': '0.1', 'incremental_inviters': '39.4', 'incremental_registrations': '51.2', 'incremental_payers': '18.0', 'incremental_revenue_30d': '125.49', 'current_20pct_obligation': '25.1', 'payer_server_cost': '4.31', 'reward_cash_cost': '5.95', 'max_reward_opportunity_cost': '56.27', 'contribution_after_max_opportunity': '33.85'}, {'scenario': 'Base', 'target_inviter_rate': '0.15', 'incremental_inviters': '96.6', 'incremental_registrations': '144.9', 'incremental_payers': '50.9', 'incremental_revenue_30d': '355.0', 'current_20pct_obligation': '71.0', 'payer_server_cost': '12.21', 'reward_cash_cost': '16.84', 'max_reward_opportunity_cost': '159.19', 'contribution_after_max_opportunity': '95.76'}, {'scenario': 'Aggressive', 'target_inviter_rate': '0.2', 'incremental_inviters': '153.8', 'incremental_registrations': '261.5', 'incremental_payers': '91.8', 'incremental_revenue_30d': '640.57', 'current_20pct_obligation': '128.11', 'payer_server_cost'

## Results

In [4]:
base = next(row for row in scenarios if row['scenario'] == 'Base')
assert 50 <= float(base['incremental_payers']) <= 52
assert float(base['contribution_after_max_opportunity']) > 0
{
    'base_incremental_payers': float(base['incremental_payers']),
    'base_incremental_revenue_30d': float(base['incremental_revenue_30d']),
    'base_contribution_after_max_opportunity': float(base['contribution_after_max_opportunity']),
    'share_of_may_to_july_gap_recovered': round(float(base['incremental_revenue_30d']) / (3581.06 - 2823.94), 3),
}

{'base_incremental_payers': 50.9, 'base_incremental_revenue_30d': 355.0, 'base_contribution_after_max_opportunity': 95.76, 'share_of_may_to_july_gap_recovered': 0.469}

## Takeaways

1. Referral quality is not the bottleneck; referral participation is.
2. A three-referral threshold is a bad launch default for Doodle because 72.4% of historical referrers stopped after one or two registrations.
3. The base scenario raises recent inviter participation from 6.6% to 15%, producing about 51 incremental payers and $356 of 30-day revenue.
4. Even after the current 20% commission and a worst-case full-month opportunity cost for every sender reward, all three scenarios remain contribution-positive before support and development.
5. The experiment should be judged on incremental first-paid referrals per eligible user and 30-day contribution, not shares or registrations.